dakara_server/playlist/consumers.py

```diff
@@ -0,0 +1,169 @@
+            self.close()
+            logger.error("Invalid user tries to connect to playlist "
+                         "device consumer")
+            return
+
+        # create the group
+        async_to_sync(self.channel_layer.group_add)(
+            self.group_name,
+            self.channel_name,
+        )
+
+        # reset current playing playlist entry if any
+        current_playlist_entry = models.PlaylistEntry.get_playing()
+        if current_playlist_entry is not None:
+            current_playlist_entry.date_played = None
+            current_playlist_entry.save()
+
+        self.accept()
+        logger.info("Player connected through websocket")
+
+    def disconnect(self, close_code):
+        async_to_sync(self.channel_layer.group_discard)(
+            self.group_name,
+            self.channel_name,
+        )
+
+        # broadcast the player is idle
+        broadcast_to_channel('playlist.front', 'send_player_idle')
+
+    def receive_ready(self, event=None):
+        """Start to play when the player is ready"""
+        # request to start playing if possible
+        logger.info("The player is ready")
+        self.handle_next()
+
+    def send_playlist_entry(self, event):
+        """Send next playlist entry
+        """
+        playlist_entry = event['playlist_entry']
+
+        if playlist_entry is None:
+            raise ValueError("Playlist entry must not be None")
+
+        # log the event
+        logger.info("The player will play '{}'".format(playlist_entry))
+
+        # send to device
+        serializer = serializers.PlaylistEntryForPlayerSerializer(
+            playlist_entry)
+        self.send_json({
+            'type': 'playlist_entry',
+            'data': serializer.data
+        })
+
+    def send_idle(self, event=None):
+        """Request the player to be idle
+        """
+        # log the event
+        logger.info("The player will play idle screen")
+
+        # send to device
+        self.send_json({
+            'type': 'idle'
+        })
+
+    def send_command(self, event):
+        """Send a given command to the player
+        """
+        command = event['command']
+
+        if command not in dict(models.Player.COMMANDS).keys():
+            raise ValueError("Unknown command requested '{}'"
+                             .format(command))
+
+        logger.info("The player will {}".format(command))
+
+        self.send_json({
+            'type': 'command',
+            'data': {
+                'command': command
+            }
+        })
+
+    def handle_next(self, event=None):
+        """Prepare the submission of a new playlist entry depending on the context
+
+        A new playlist entry will be sent to the player if:
+            - the kara status is in play mode;
+            - there is a new playlist entry in playlist after the provided one.
+        """
+        # get the next playlist entry if the kara is in play mode
```

Comment:
Misleading comment here. Should be "request to be idle if the kara is not in play mode".

Reply:
I agree
```suggestion
        # request to be idle if the kara is not in play mode
```